[WIP] Granular Ammo Bench Categorization

Hatterhat opened the pull request:
## About The Pull Request
Opening the PR to see the line count.
Crux of the matter:
- Ammo benches now use bitflags to determine what ammo types they can print.
- To accomodate for this, ammo casings now have bitflags matching benches' bitflags.
- If a casing matches the bench's bitflags, it's valid to print. If not, it's invalid to print.

PR's not done but we'll get there, trust.
## How This Contributes To The Nova Sector Roleplay Experience
Ammo types funny. Balancing ammo availability even funnier.

## Proof of Testing
We'll burn that bridge when we get to it.

## Changelog

:cl:
balance: Shuffles around so, so much of the ammunition workbench. If this gets merged while this changelog is unwritten, yell at a maintainer.
balance: Due to lobbying by defense and import companies, ammunition workbenches now work on a more granular licensing system, dictating what ammo they can or cannot print based on what license keys have been uploaded.
/:cl:

OrionTheFox commented on modular_nova/modules/ammo_workbench/code/ammo_workbench.dm:
```diff
@@ -52,11 +43,10 @@
 
 /obj/item/circuitboard/machine/ammo_workbench
-	name = "Ammunition Workbench (Machine Board)"
+	name = "Ammunition Workbench"
```
```suggestion
	name = "ammunition workbench"
```
Super small nit and barely related to the actual content but might as well catch it now - item names should in most circumstances be lowercase

Rest of the PR is both blessed and cursed. All I can say is Godspeed and good luck.

thetaro commented:
is there like a design doc for this because i still dont understand how it actually works and you didnt really describe what it does, it just sounds like ammo printing will be wholly useless and you're better off buying ammo, because why struggle with needing tech and only having a limited amount printable when 50 credits can buy you a mag (or more) and you can bulk buy

RealWinterFrost commented:
Im loving the new system! I've noticed 3 boards or 2 may be of different names or even just copies of each other, as one of them is a carbon copy of another. The UI is a bit confusing with the [0] and [1]. May I suggest adding a suffix after like [1 Point(s)] [0 Point(s)]?

annikaRU commented:
Everything is fine besides the points, as a cargo main mostly QM, the points just leaves a really foul aftertaste given you have to constantly rebuy modules and most people have tended to agree on this

Hatterhat commented:
> Everything is fine besides the points, as a cargo main mostly QM, the points just leaves a really foul aftertaste given you have to constantly rebuy modules and most people have tended to agree on this, even a method to just bypass this would be preferable.

the points are intentionally put there as a check for Printing Ten Bajillion Armor Piercing

RealWinterFrost commented:
I think boosting the amount of points available on all of them about a third (1/3) more would be more comfortable as to not constantly reauthenticate them giving them a slight buff!. But I like them as of the current moment!

stobarico commented:
Having to not only spend thousands on non-descriptive boards which have an extremely limited artificial capacity, and having to switch them out everytime you want a different ammo is abhorent, and feels horrible to play. 

There is a balance between balance, fun and realism, this is absurdly unfun, and doesn't even do that much, as you can just buy the ammo from cargo instead of printing it, there must be a better way to nerf ballistics if this is the intention of this PR.

Hatterhat commented:
- Reauthenticators now just work on a point system and not an arbitrary reset.
- Doubled the baseline capacity of ammo modules.
- Added one (1) basic lethal module to the SecVend.
- Reduced prices of ammo bench related miscellany.
    - Ammo bench is 200cr.
    - Basic lethal is 300cr.
    - Bulk basic lethal is 400cr.
    - Niche lethal (TRAC, stardust, EMP, incendiary, etc.) is 500cr.
    - Variant lethal (HP/AP) is 800cr.
- Added milspec buckshot, milspec slugs, and the evil ammo printer module to the Syndicate uplink.

annikaRU commented:
A minor suggestion is for EMAGs to disable the point function if people don't wish to deal with it constantly they can buy a binyat or somehow convince a traitor to emag it

Hatterhat commented:
- Flattens AP and HP categories into "plus", which the variant module still prints.
- Kills off the bulk lethal module, considering there's more availability of ammo bench.
- Kills off "smart" and "thermal", rolling them into "niche".
- Ammo modules now have color-coded stripes.
- Security can now order an ammo bench starter kit with some modules, rebooters, and another flatpacked ammo bench.
- Makes the ammo bench UI hide materials it doesn't have, condenses the module tab into the main tab, and adds an instruction guide.
<details>

![image](https://github.com/user-attachments/assets/3f52aaba-1b9f-43a3-be38-cc7c25cc8147)
![image](https://github.com/user-attachments/assets/109c4737-6f4a-424f-8f49-91a30a5e7db4)

</details>

Hatterhat commented on modular_nova/modules/modular_weapons/code/modular_projectiles/pistol_calibers.dm:
```diff
@@ -0,0 +1,247 @@
+	projectile_type = /obj/projectile/bullet/c45/rubber
+	ammo_categories = AMMO_CLASS_NONE
+	harmful = FALSE
+	print_cost = 0
```
I think there *is* a case for non-harmful but still cost-required bullets that have niches or are otherwise higher-performing than conventional ammunition such as .38 TRAC (low damage, but leaves a tracking implant in people so you can continuously hunt down whoever you tagged with it) or stardust shells (it's a taser in a shotgun shell, for better or for worse)? Pretty sure this is something covered under the NICHE_LTL category or something like that.
